plugins/extractit/icon_extractit/util/extractor.py

```diff
@@ -32,16 +35,28 @@ def extract_filepath(provided_regex: str, provided_string: str, provided_file: s
 def strip_subdomains(matches: list) -> list:
     for match in enumerate(matches):
         stripped_domain = tldextract.extract(match[1])
-        matches[match[0]] = ".".join(stripped_domain[1:3])
+        # In some cases, tldextract recognizes a suffix as a domain, adds the domain to subdomain, and returns an empty
+        # string as suffix, so we check that tldextract extracted the suffix.
+        if not stripped_domain.suffix:
+            suffix = stripped_domain.domain
+            subdomain = stripped_domain.subdomain
+            if subdomain and suffix:
+                # here we split `subdomain` and extract the domain, which is the last element from the `subdomains` list
+                subdomains = subdomain.split(".")
+                matches[match[0]] = f"{subdomains[len(subdomains) - 1]}.{suffix}"
+        else:
+            matches[match[0]] = ".".join(stripped_domain[1:3])
     return list(dict.fromkeys(matches))
 
 
 def clear_domains(matches: list) -> list:
     new_matches = []
     for match in enumerate(matches):
         if not match[1].endswith("@"):
-            new_matches.append(match[1].split("/")[0])
-    return new_matches
+            split_match = match[1].split("/")[0]
+            if not split_match.endswith("="):
+                new_matches.append(split_match)
+    return list(dict.fromkeys(new_matches))
```

Comment:
If I understand this correctly, we are stripping away the parameters from the URL by looking for values with `=`. Could we not just strip away everything that comes after the `?` in the URL? We know that in a URL, parameters are anything that comes after the `?` symbol.
Do we want some of the domains in the parameters?
I think we could use some more comments & documentation explaining what this function does, thank you.